banyand/tsdb/index/writer.go

```diff
@@ -76,36 +77,37 @@ func NewWriter(ctx context.Context, options WriterOptions) *Writer {
 	}
 	w.shardNum = options.ShardNum
 	w.db = options.DB
+	w.enableGlobalIndex = options.EnableGlobalIndex
 	w.indexRuleIndex = partition.ParseIndexRuleLocators(options.Families, options.IndexRules)
-	w.ch = run.NewChan[Message](make(chan Message))
+	w.ch = make(chan Message)
 	w.bootIndexGenerator()
 	return w
 }
 
 func (s *Writer) Write(value Message) {
 	go func(m Message) {
-		s.ch.Write(m)
+		s.ch <- m
 	}(value)
 }
 
 func (s *Writer) Close() error {
-	return s.ch.Close()
```

Comment:
Since we removed `Close()` here, is it possible we still have many goroutines waiting to write to the channel?

Reply:
We can use close to notify the receivers, not the senders. The senders will be waiting to write. That is the purpose of this process.

Reply:
> We can use close to notify the receivers, not the senders. The senders will be waiting to write. That is the purpose of this process.

Yes, we can use some stop channel to notify the receivers, but what about the senders?

> They won't block the GC from collecting them.

From this issue, https://github.com/golang/go/issues/19702, Go may not be clever enough 
 now to collect blocking goroutines.

Reply:
Automatically closing always has this kind of bugs in GC phases.
Be careful. We don't want a leak 😄